data/hooks/diagnosis/10-ip.py

```diff
@@ -220,9 +217,9 @@ def get_public_ip(self, protocol=4):
         try:
             return download_text(url, timeout=30).strip()
         except Exception as e:
-            self.logger_debug(
-                "Could not get public IPv%s : %s" % (str(protocol), str(e))
-            )
+            protocol = str(protocol)
+            e = str(e)
+            self.logger_debug(f"Could not get public IPv{protocol} : {e}")
```

Comment:
IIRC you can do something like that:

```suggestion
            self.logger_debug(f"Could not get public IPv{str(protocol)} : {str(e)}")
```

Reply:
Ah yes, in fact just using `{protocol}` probably cast it as a string, but i didn't want to introduce any regression and was too lazy to confirm it so idk wat do é_è

Reply:
it was more a general remark, not specific to this line :)